File: qqbot-plugin-api/src/main/java/com/mieai/qqbot/plugin/api/MessageTarget.java
package com.mieai.qqbot.plugin.api;

import static java.util.Objects.requireNonNull;

/** A QQ target identifier; the target type prevents cross-scope sends. */
public record MessageTarget(MessageTargetType type, String id) {
    public MessageTarget {
        requireNonNull(type, "type must not be null");
        requireNonNull(id, "id must not be null");
        if (id.isBlank() || !id.equals(id.strip()) || id.codePoints().anyMatch(Character::isWhitespace)) {
            throw new IllegalArgumentException("id must be a non-blank token");
        }
        if (id.length() > 255) throw new IllegalArgumentException("id is too long");
    }
}

File: qqbot-plugin-api/src/main/kotlin/com/mieai/qqbot/plugin/api/MessageTarget.kt
package com.mieai.qqbot.plugin.api

/** A QQ target identifier; the target type prevents cross-scope sends. */
data class MessageTarget(
    val type: MessageTargetType,
    val id: String,
) {
    init {
        require(id.isNotBlank() && id == id.trim() && id.codePoints().noneMatch { Character.isWhitespace(it) }) {
            "id must be a non-blank token"
        }
        require(id.length <= 255) { "id is too long" }
    }
}
